stubs/Symfony/Component/PropertyAccess/PropertyAccessorInterface.stub

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Symfony\Component\PropertyAccess;
+
+interface PropertyAccessorInterface
+{
+
+    /**
+     * @phpstan-template T of object|array<mixed>
+     * @phpstan-param T &$objectOrArray
+     * @phpstan-param-out ($objectOrArray is object ? T : array<mixed>) $objectOrArray
```

Comment:
hi, just for my understanding - this could be simply `@phpstan-param-out T $objectOrArray` as well, right?

I had a local stub with something like that and got an error when updating -> could remove my local stub -> thx! :)

Reply:
```
@phpstan-template T of object|array<mixed>
@phpstan-param T &$objectOrArray
@phpstan-param-out T $objectOrArray
```
doesn't give the same beahvior

It fails 
```
$array = [1 => 'ea'];
$propertyAccessor->setValue($array, 'foo', 'bar');
assertType('array', $array);
```
You'll get `array{1: 'ea'}` instead, which is wrong.

Reply:
Ah OK, makes sense. Thx!